File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
    const max = Math.max(...candies)
    const new_arr = [];

    for (let i = 0; i < candies.length; i++){
        if(candies[i] + extraCandies >= max){
            new_arr.push(true);
        } else {
            new_arr.push(false);
        };
    };
    return new_arr;
};

File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        n = len(candies)
        result = [False]*n
        minCandies = max(candies) - extraCandies
        
        for i in range(n):
            if candies[i] >= minCandies:
                result[i] = True

        return result
        